CHANGELOG.md

```diff
@@ -0,0 +1,37 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
+
+## [Unreleased]
+
+### Added
+
+- Added `n_points` parameter to all functions in `shap.datasets` (#39).
+- Added the `ruff` linter (#25, #26, #27).
+
+### Fixed
+
+- Fixed failing unit tests (#29, #20, #24).
+- Fixed `plot.waterfall` to support yticklabels with boolean features (#58).
+- Prevent `TreeExplainer.__call__` from throwing ValueError when passed a pandas DataFrame containing Categorical columns (#88).
+- Fixed sampling in `shap.datasets` to sample without replacement (#36).
+- Fixed deprecation warnings for `numpy>=1.24` from numpy types (#7).
+- Fixed deprecation warnings for `Ipython>=8` from `Ipython.core.display` (#13).
+- Fixed deprecation warnings for `tensorflow>=2.11` from `tf.optimisers` (#16).
+- Fixed deprecation warnings for `sklearn>=1.2` from `sklearn.linear_model` (#22).
+- Fixed installation of package via setuptools (#51).
+
+### Changed
+
+
+### Removed
+
+- Deprecated the Boston house price dataset (#38).
+- Removed the unused `mimic.py` file and `MimicExplainer` code (#53).
```

Comment:
The tagged issue and PR numbers are for dsgibbons/shap. Do we need to add an identifier to reference the fork explicitly?

Reply:
Good suggestion, I'll do that now. We will probably want to overhaul the changelog in near future, e.g. host as part of the docs.

Reply:
yep, I was just about to comment this as well. I'll tag along here:

---

What do we think about all of these PR references in the CHANGELOG? They're no longer correct once we merge it into `slundberg/shap`. Shall we prepend the repo name `dsgibbons#xx` to all of these? And also this PR number?

How about something like

```
- Removed the unused `mimic.py` file and `MimicExplainer` code (dsgibbons#53, #2943).
```
?

This ensures everyone can locate the original PR where the discussions happened (even if/after we migrate to the organization)

Reply:
> How about something like
> ...

Looks good!

Reply:
I tried adding full URLs, how does that look to you?

```markdown
- Deprecated the Boston house price dataset
  ([#38](https://github.com/dsgibbons/shap/pull/38)).
- Removed the unused `mimic.py` file and `MimicExplainer` code
  ([#53](https://github.com/dsgibbons/shap/pull/53)).
```
  
Renders as:
  
- Deprecated the Boston house price dataset ([#38](https://github.com/dsgibbons/shap/pull/38)).
- Removed the unused `mimic.py` file and `MimicExplainer` code ([#53](https://github.com/dsgibbons/shap/pull/53)).
  
FWIW, I think we will need to tweak the changelog more in future so happy to park any further adjustments to another PR

Reply:
Any further comments on this over at https://github.com/dsgibbons/shap/pull/105